src/platform_linux/shmem.c

```diff
@@ -517,11 +503,12 @@ platform_shm_alloc(platform_heap_id hid,
    }
 
    // Optimistically, allocate the requested 'size' bytes of memory.
-   retptr = __sync_fetch_and_add(&shminfop->shm_next, size);
+   _Static_assert(sizeof(void*)==sizeof(size_t), "check our casts are valid");
```

Comment:
It's ok to add it here; I understand why you are doing it. But, it's a very low-level assertion and I think it probably belongs elsewhere in `platform.c`.

You may choose to not do anything right now as this is proto code. I think it would be better to create a new function in `platform.c` named:

`void __attribute__((constructor)) platform_asserts()`

and stuff all such platform-wide asserts in one place.

(This is just a suggestion; feel free to file it away for a future date.)

Reply:
I'm going to leave this as-is for now.  We can return to it if/when we're prepping this branch for integration to `main`